pkg/cluster/cert.go

```diff
@@ -228,6 +193,33 @@ func recreateSecret(ctx context.Context, c client.Client, existingSecret, newSec
 	return nil
 }
 
+// generateCertSecret creates a secret if it does not exist;  recreate this secret if type not match; update data if outdated.
+func generateCertSecret(ctx context.Context, c client.Client, certSecret *corev1.Secret, secretName, namespace string) error {
+	existingSecret := &corev1.Secret{}
+	err := c.Get(ctx, client.ObjectKey{Name: secretName, Namespace: namespace}, existingSecret)
+	if err != nil {
```

Comment:
I find this chain of if if if else elseif if a bit more confusing than the proposal, but fine for me if it's better for you, clear enough.

Reply:
My way of thinking behind using switch is that we have 3 cases for Get: existing object, non-existing object, error of fetching. Not 2 as for if-else.

Reply:
i was not very fond of using switch in this case, thats why tried to avoid it.
but after reading Luca's comments, maybe switch is much better for a quick return. 
let me do some updates again